stm/stm.ml

```diff
@@ -2681,7 +2681,9 @@ let process_transaction ~doc ?(newtip=Stateid.fresh ())
       | VtStartProof (guarantee, names) ->
 
          if not (get_allow_nested_proofs ()) && VCS.proof_nesting () > 0 then
-           "Nested proofs are not allowed unless you turn the Nested Proofs Allowed flag on."
+           "Nested proofs are discouraged and not allowed by default. This error probably means that you forgot \
+           to close forgot to close the last \"Proof.\" with \"Qed.\" or \"Defined.\". \
+           If you really intended to use nested proofs, you can do so by turning the \"Nested Proofs Allowed\" flag on."
            |> Pp.str
```

Comment:
Should this become a `strbrk`?

Reply:
I changed the three things you mentioned. Whats the difference? (the comment above strbrk says "to remove" btw)

Reply:
IIRC strbrk consider each space a potential breaking point, while str should be printed all on the same line

Reply:
Oh, and just for your info, in recent ocaml you can `{| multi line string with no need to escape " |}`

Reply:
(but here indentation may get in the way...)